nixos/modules/services/web-apps/hledger-web.nix

```diff
@@ -26,19 +26,46 @@ in {
       '';
     };
 
-    capabilities = mkOption {
-      type = types.commas;
-      default = "view";
+    capabilities = {
+      view = mkOption {
+        type = types.bool;
+        default = true;
+        description = ''
+          Enable the view capability.
+        '';
+      };
+      add = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Enable the add capability.
+        '';
+      };
+      manage = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Enable the manage capability.
+        '';
+      };
+    };
+
+    stateDir = mkOption {
+      type = types.path;
+      default = "/var/lib/hledger-web";
       description = ''
-        Enable the view, add, and/or manage capabilities. E.g. view,add
+        Path the service has access to. If left as the default value this
+        directory will automatically be created before the hledger-web server
+        starts, otherwise the sysadmin is responsible for ensuring the
+        directory exists with appropriate ownership and permissions.
       '';
     };
 
     journalFile = mkOption {
-      type = types.path;
-      example = "/home/hledger/.hledger.journal";
+      type = types.str;
+      default = ".hledger.journal";
```

Comment:
Somehow this breaking change slipped by me before. You might consider adding a `check` or `assertion` that the value here is a file name and not a path+file name. Release notes wouldn't hurt either...

You could probably do better, but at least checking that the name doesn't start with a forward slash could help :man_shrugging:

Reply:
I guess that would make sense to have. I implemented an assertion:

```
error:                                                                                                                               
       Failed assertions:                                                                                                            
       - services.hledger-web.journalFile is no longer defined as a path, but                                                        
       as a file name located in services.hledger-web.stateDir.                                                                      
```

Given that `hledger` wasn't mentioned once in the NixOS release notes, I'd figure the module isn't used enough to justify an entry?